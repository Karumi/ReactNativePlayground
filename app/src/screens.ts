const screens = {
    resources: "Resources",
    navigation: "Navigation",
    redux: "Redux",
    layouts: "Layouts",
    navigationModal: "NavigationModalScreen",
    list: "ListOfComponents",
    scroll: "ScrollViewComponent",
    grid: "GridViewComponent",
    camera: "CameraComponent",
    imagePicker: "ImagePickerComponent",
    mapScreen: "MapScreen",
};

export default screens;
