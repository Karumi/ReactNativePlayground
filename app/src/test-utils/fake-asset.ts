jest.mock("../assets/asset", () => {
return {
        default: (uri) => uri,
    };
});
