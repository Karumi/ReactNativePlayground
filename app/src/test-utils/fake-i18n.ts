
import { I18nResolver } from "i18n-ts";
import en from "../translations/en";

const i18n = { default: en };

jest.mock("../i18n", () => {
return {
        default: () => new I18nResolver(i18n,  "en").translation,
    };
} );
