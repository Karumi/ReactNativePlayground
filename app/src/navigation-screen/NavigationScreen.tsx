import { Button, Container, Text } from "native-base";
import * as React from "react";
import { Alert } from "react-native";
import Toolbar from "../base-components/toolbar/Toolbar";
import translator from "../i18n";
import screens from "../screens";

export interface Props {
    navigation: any;
}
class NavigationScreen extends React.Component<Props> {
    public render() {
        return (
            <Container>
                <Toolbar
                    title={translator().navigationScreenTitle}
                    navigation={this.props.navigation}
                    showBackButton={true}
                />
                <Button style={{ margin: 10 }} onPress={() => this.open(screens.resources)}>
                    <Text>Open resources screen</Text>
                </Button>
                <Button style={{ margin: 10 }} onPress={() => this.goBack()}>
                    <Text>Go back</Text>
                </Button>
                <Button style={{ margin: 10 }} onPress={() => this.showDialog()}>
                    <Text>Open dialog</Text>
                </Button>
                <Button style={{ margin: 10 }}>
                    <Text>Open modal screen</Text>
                </Button>
            </Container>
        );
    }

    private open(screen: string) {
        this.props.navigation.navigate(screen);
    }

    private goBack() {
        this.props.navigation.pop();
    }

    private showDialog() {
        Alert.alert(
            "Alert Title",
            "My Alert Msg",
            [
                { text: "Ask me later", onPress: () => alert("Ask me later pressed") },
                { text: "Cancel", onPress: () => alert("Cancel Pressed"), style: "cancel" },
                { text: "OK", onPress: () => alert("OK Pressed") },
            ],
            { cancelable: false },
        );
    }
}

export default NavigationScreen;
