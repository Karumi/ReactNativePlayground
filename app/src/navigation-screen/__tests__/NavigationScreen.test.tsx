import { Button } from "native-base";
import * as React from "react";
import * as renderer from "react-test-renderer";
import showAlertDialog from "../../base-components/alert-dialog/alert";
import screens from "../../screens";
import "../../test-utils/fake-i18n";
import NavigationScreen from "../NavigationScreen";

jest.mock("../../base-components/toolbar/Toolbar", () => {
    return {
        default: "Toolbar",
    };
});

jest.mock("../../base-components/alert-dialog/alert", () => {
    return {
        default: jest.fn(),
    };
});

const navigationWithoutParam = {
    getParam() {
        return null;
    },
};

const navigationWithParam = {
    getParam: jest.fn(),
    navigate: jest.fn(),
    pop: jest.fn(),
};

it("renders correctly with defaults", () => {
    const screen = renderer.create(<NavigationScreen navigation={navigationWithoutParam} />).toJSON();
    expect(screen).toMatchSnapshot();
});

it("renders correctly with a text param passed as argument", () => {
    const anyText = "React native rules";
    givenTheScreenIsOpenWithTextParamEqualTo(anyText);

    const screen = renderer.create(<NavigationScreen navigation={navigationWithParam} />).toJSON();

    expect(screen).toMatchSnapshot();
});

it("opens the resources screen when the user taps on the first button", () => {
    const screen = renderer.create(<NavigationScreen navigation={navigationWithParam} />).root;

    tapOnButtonNumber(screen, 0);

    expect(navigationWithParam.navigate).toHaveBeenCalledWith(screens.resources);
});

it("opens the previous screen when the user taps on the second button", () => {
    const screen = renderer.create(<NavigationScreen navigation={navigationWithParam} />).root;

    tapOnButtonNumber(screen, 1);

    expect(navigationWithParam.pop).toHaveBeenCalled();
});

it("opens the alert dialog when the user taps on the third button", () => {
    const screen = renderer.create(<NavigationScreen navigation={navigationWithParam} />).root;

    tapOnButtonNumber(screen, 2);

    expect(showAlertDialog).toHaveBeenCalled();
});

it("opens the navigation screen in modal mode when the user taps on the fourth button", () => {
    const screen = renderer.create(<NavigationScreen navigation={navigationWithParam} />).root;

    tapOnButtonNumber(screen, 3);

    expect(navigationWithParam.navigate).toHaveBeenCalledWith(screens.navigationModal, { text: "Modal Mode" });
});

function tapOnButtonNumber(screen, buttonIndex: number) {
    screen.findAllByType(Button)[buttonIndex].instance.props.onPress();
}

function givenTheScreenIsOpenWithTextParamEqualTo(text: string) {
    navigationWithParam.getParam.mockReturnValue(text);
}
