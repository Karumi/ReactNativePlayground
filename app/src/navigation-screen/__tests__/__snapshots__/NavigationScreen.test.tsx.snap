// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with a text param passed as argument 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <Toolbar
    navigation={
      Object {
        "getParam": [MockFunction] {
          "calls": Array [
            Array [
              "text",
              "",
            ],
          ],
          "results": undefined,
        },
        "navigate": [MockFunction],
        "pop": [MockFunction],
      }
    }
    showBackButton={true}
    title="Navigation"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#007aff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "margin": 10,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
    >
      Open resources screen
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#007aff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "margin": 10,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
    >
      Go back
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#007aff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "margin": 10,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
    >
      Open dialog
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#007aff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "margin": 10,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
    >
      Open modal screen
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    uppercase={false}
  >
    React native rules
  </Text>
</View>
`;

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <Toolbar
    navigation={
      Object {
        "getParam": [Function],
      }
    }
    showBackButton={true}
    title="Navigation"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#007aff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "margin": 10,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
    >
      Open resources screen
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#007aff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "margin": 10,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
    >
      Go back
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#007aff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "margin": 10,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
    >
      Open dialog
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#007aff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "margin": 10,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
    >
      Open modal screen
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    uppercase={false}
  />
</View>
`;
